File: composeApp/src/androidMain/kotlin/com/prekogdevs/fintrack/feature/dashboard/components/BalanceCard.kt
package com.prekogdevs.fintrack.feature.dashboard.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.prekogdevs.fintrack.core.theme.AppColors

@Composable
fun BalanceCard(totalBalance: Double, income: Double, expenses: Double) {
    Box(
        modifier = Modifier
            .padding(horizontal = 20.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(
                Brush.horizontalGradient(listOf(AppColors.BlueDark, AppColors.BlueLight))
            )
            .padding(24.dp)
    ) {
        Column {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("$  ", color = Color.White.copy(alpha = 0.8f), fontSize = 14.sp)
                Text("Total Balance", color = Color.White.copy(alpha = 0.8f), fontSize = 14.sp)
            }
            Spacer(Modifier.height(8.dp))
            Text(
                text = "$${"%.2f".format(totalBalance)}",
                fontSize = 36.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White
            )
            Spacer(Modifier.height(20.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                BalanceStat(
                    label = "Income",
                    amount = income,
                    isIncome = true,
                    modifier = Modifier.weight(1f)
                )
                BalanceStat(
                    label = "Expenses",
                    amount = expenses,
                    isIncome = false,
                    modifier = Modifier.weight(1f)
                )
            }
        }
    }
}

File: iosApp/iosApp/Feature/Dashboard/Components/BalanceCard.swift
import SwiftUI

struct BalanceCard: View {
    let totalBalance: Double
    let income: Double
    let expenses: Double

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 2) {
                Text("$  ")
                    .foregroundColor(.white.opacity(0.8))
                    .font(.system(size: 14))
                Text("Total Balance")
                    .foregroundColor(.white.opacity(0.8))
                    .font(.system(size: 14))
            }

            Text(String(format: "$%.2f", totalBalance))
                .font(.system(size: 36, weight: .bold))
                .foregroundColor(.white)

            Spacer().frame(height: 12)

            HStack(spacing: 12) {
                BalanceStat(label: "Income", amount: income, isIncome: true)
                BalanceStat(label: "Expenses", amount: expenses, isIncome: false)
            }
        }
        .padding(24)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            LinearGradient(
                colors: [AppColors.blueDark, AppColors.blueLight],
                startPoint: .leading,
                endPoint: .trailing
            )
        )
        .cornerRadius(20)
        .padding(.horizontal, 20)
    }
}
